resources/page/permalinks.go

```diff
@@ -333,6 +362,36 @@ func (l PermalinkExpander) translationBaseName(p Page) string {
 	return p.File().TranslationBaseName()
 }
 
+// withSectionPagesFunc returns a function that builds permalink attributes from section pages.
+// It applies the transformation function f to each ancestor section (Page), then joins the results with the join function.
+//
+// Current use is create section-based hierarchical paths using section slugs.
+func (l PermalinkExpander) withSectionPagesFunc(f func(Page, string) (string, error), join func(...string) string) func(p Page, s string) (string, error) {
+	return func(p Page, s string) (string, error) {
+		var entries []string
+		sectionEntries := p.CurrentSection().SectionsEntries()
+
+		for i := range sectionEntries {
```

Comment:
While this would, it's much more effective to just do `for _, section := range p.Ancestors().Reverse() {}`